src/client/ui/Button.ts

```diff
@@ -0,0 +1,185 @@
+import * as PIXI from "pixi.js";
+import { ButtonState } from "./ButtonState.ts";
+import { gameConfig } from "../config/GameConfig.ts";
+
+export type stateUI = {
+  backgroundColor: number;
+  textLabel: string;
+  state: ButtonState;
+};
+
+export class Button extends PIXI.Container {
+  private background = new PIXI.Graphics();
+
+  private text = new PIXI.Text();
+
+  private _clickEvent: () => void;
+  private _pointerOver: () => void;
+  private _pointerOut: () => void;
+
+  private defaultUI: stateUI = {
+    backgroundColor: gameConfig.backgroundDefaultColor,
+    textLabel: "",
+    state: ButtonState.ready,
+  };
+
+  private buttonUI: stateUI = this.defaultUI;
+
+  private _buttonUIReady: stateUI = {
+    backgroundColor: gameConfig.backgroundDefaultColor,
+    textLabel: "",
+    state: ButtonState.ready,
+  };
+  private _buttonUIDisabled: stateUI = {
+    backgroundColor: gameConfig.backgroundDefaultColor,
+    textLabel: "",
+    state: ButtonState.disabled,
+  };
+  private _buttonUIPointerHover: stateUI = {
+    backgroundColor: gameConfig.backgroundDefaultColor,
+    textLabel: "",
+    state: ButtonState.pointerhover,
+  };
+
+  private _buttonState: ButtonState = ButtonState.ready;
+
+  buttonWidth: number;
+  buttonHeight: number;
+  buttonX: number;
+  buttonY: number;
+
+  constructor(
+    width: number,
+    height: number,
+    x: number,
+    y: number,
+    textLabel: string = "click",
+    clickEvent: () => void = () => {},
```

Comment:
instead of empty functions you may want to use nullable type, and optional chaining for function call 
`clickEvent?.();`